apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxScreen.tsx

```diff
@@ -149,78 +207,49 @@ export const ConfirmTxScreen = () => {
           </>
         }
         onClose={() => {
-          setConfirmationModal(false)
+          setScreenState({...screenState, modal: false, swapTx: undefined})
         }}
         containerStyle={{justifyContent: 'space-between'}}
       />
-
-      <BottomSheetModal
-        isOpen={bottomSheetState.isOpen}
-        title={bottomSheetState.title}
-        content={<Text style={styles.text}>{bottomSheetState.content}</Text>}
-        onClose={() => {
-          setBottomSheetSate({isOpen: false, title: '', content: ''})
-        }}
-      />
     </SafeAreaView>
   )
 }
 
 const Actions = ({style, ...props}: ViewProps) => <View style={[styles.actions, style]} {...props} />
 
+export const useSwapTx = (
```

Comment:
on `ConfirmTx` the unsignedTx should be created cuz it needs to present the summary, this should happen before navigating/opening the confirm screen

Reply:
still now sure about this as I have taken the example from `WithdrawStakingRewards` There we create the `unsignedTx ` only when showing the modal/popup for the user to sign it 🤔  in Swap -> On summary right now we show only the tokens (buy/sell) info and fees - then when user presses confirm we get unsignedTx and maybe show other info in the modal